backend/src/services/generateSessionAccessToken.ts

```diff
@@ -0,0 +1,34 @@
+import { getSessionAccessTokenByTokenValue } from "../data/sessionsDataAccess";
+
+export async function generateSessionAccessToken(): Promise<string> {
+  // this returns a unique session token. It will re-try until it finds a unique one
+  // that does not exist in the db yet before returning the value.
+
+  let sessionToken: string = "";
+
+  let run = true;
+
+  while (run) {
```

Comment:
In don't think it's responsability of this function to check wether the code exists in the db. It's role should be just to generate a random code. (If you use a library, this function is not needed anymore).

I would move that logic in `createSessionAccessToken`, and instead of a possibile infite loop, I would use a "retries" approach. In order for it to work, the code must be a unique key in the db.
This way we have only one write operation instead of a read and than a write each time we generate a code.

```
async function generateSessionAccessToken() {
    const MAX_RETRIES = 3;

    for (let i = 0; i < MAX_RETRIES; i++) {
        try {
            // generate code
            // try to save code to db
            // return code
        } catch (e) {}
    }
    //throw an error or return null
}
 ```

Reply:
That makes a lot of sense. Thanks Moh